tests/MongoDB.Driver.Tests/Specifications/crud/FindOneAndUpdateTest.cs

```diff
@@ -81,11 +88,31 @@ protected override BsonDocument ExecuteAndGetResult(IMongoCollection<BsonDocumen
         {
             if (async)
             {
-                return collection.FindOneAndUpdateAsync(_filter, _update, _options).GetAwaiter().GetResult();
+                if (_updatePipeline != null)
+                {
+                    var pipeline = PipelineDefinition<BsonDocument, BsonDocument>.Create(_updatePipeline.Select(c => c.AsBsonDocument));
+                    return collection
+                        .FindOneAndUpdateAsync(_filter, pipeline, _options).GetAwaiter()
+                        .GetResult();
+                }
+                else
+                {
+                    return collection
+                        .FindOneAndUpdateAsync(_filter, _update, _options).GetAwaiter()
+                        .GetResult();
+                }
             }
             else
             {
-                return collection.FindOneAndUpdate(_filter, _update, _options);
+                if (_updatePipeline != null)
+                {
+                    var pipeline = PipelineDefinition<BsonDocument, BsonDocument>.Create(_updatePipeline.Select(c => c.AsBsonDocument));
+                    return collection.FindOneAndUpdate(_filter, pipeline, _options);
+                }
+                else
+                {
+                    return collection.FindOneAndUpdate(_filter, _update, _options);
+                }
```

Comment:
These lines can remain (almost) the same as before if you move the work of converting the `_update` value to an `UpdateDefinition` outside the `if` statement. It's the same for both branches anyway.